lib/sles4sap/aws_cli.pm

```diff
@@ -0,0 +1,495 @@
+    );
+    assert_script_run("aws ec2 create-tags --resources $sg_id --tags Key=OpenQAJobSg,Value=$job_id --region $region");
+    return $sg_id;
+}
+
+=head2 aws_get_security_group_id
+
+    my $sg_id = aws_get_security_group_id($region, $job_id);
+
+Retrieve the security group ID associated with a specific OpenQA job
+
+=over
+
+=item B<$region> - AWS region where the security group is located
+
+=item B<$job_id> - OpenQA job identifier used to tag the security group
+
+=back
+
+Returns the security group ID
+=cut
+
+sub aws_get_security_group_id {
+    my ($region, $job_id) = @_;
+    return script_output(
+        "aws ec2 describe-security-groups --filters 'Name=tag:OpenQAJobSg,Values=$job_id' --region $region" .
+          " --query 'SecurityGroups[0].GroupId' --output text");
+}
+
+=head2 aws_create_subnet
+
+    my $subnet_id = aws_create_subnet($region, $cidr, $vpc_id);
+
+Create a subnet within a VPC with a specified CIDR block and tag it with the OpenQA job ID
+
+=over
+
+=item B<$region> - AWS region where to create the subnet
+
+=item B<$cidr> - CIDR block for the subnet (e.g., '10.0.1.0/24')
+
+=item B<$vpc_id> - ID of the VPC where the subnet will be created
+
+=item B<$job_id> - OpenQA job identifier used to tag the security group
+
+=back
+
+Returns the subnet ID
+=cut
+
+sub aws_create_subnet {
+    my ($region, $cidr, $vpc_id, $job_id) = @_;
+    my $subnet_id = script_output(
+        "aws ec2 create-subnet --region $region --cidr-block $cidr --vpc-id $vpc_id --query 'Subnet.SubnetId' --output text");
+    assert_script_run("aws ec2 create-tags --resources $subnet_id --tags Key=OpenQAJobSubnet,Value=$job_id --region $region");
+    return $subnet_id;
+}
+
+=head2 aws_get_subnet_id
+
+    my $subnet_id = aws_get_subnet_id($region, $job_id);
+
+Retrieve the subnet ID associated with a specific OpenQA job
+
+=over
+
+=item B<$region> - AWS region where the subnet is located
+
+=item B<$job_id> - OpenQA job identifier used to tag the subnet
+
+=back
+
+Returns the subnet ID
+=cut
+
+sub aws_get_subnet_id {
+    my ($region, $job_id) = @_;
+    return script_output(
+        "aws ec2 describe-subnets --filters 'Name=tag:OpenQAJobSubnet,Values=$job_id' --region $region --query 'Subnets[0].SubnetId' --output text");
+}
+
+=head2 aws_create_internet_gateway
+
+    my $igw_id = aws_create_internet_gateway($region);
+
+Create an internet gateway and tag it with the OpenQA job ID
+
+=over
+
+=item B<$region> - AWS region where to create the internet gateway
+
+=item B<$job_id> - OpenQA job identifier used to tag the security group
+
+=back
+
+Returns the internet gateway ID
+=cut
+
+sub aws_create_internet_gateway {
+    my ($region, $job_id) = @_;
+    my $igw_id = script_output("aws ec2 create-internet-gateway --region $region --query 'InternetGateway.InternetGatewayId' --output text", 60);
+    assert_script_run("aws ec2 create-tags --resources $igw_id --tags Key=OpenQAJobIgw,Value=$job_id --region $region");
```

Comment:
Now I get there's a pattern... each of these function is creating a resource and tag it in a two commands approach. Maybe you can create a function to tag and use it everywhere